### PR TITLE
Use pytest

### PR DESCRIPTION
**Pull request by rc**
This PR aims to address #363 and #552.

As a proof of concept, run:
```
pytest -sv tests/test_assembling.py
pytest -sv tests/test_base.py
```
I will convert more test files as time permits. @vlukes, @wd15, if you have some recommendations, please chime in. I will probably move the `tests/` directory into `sfepy/`, as mentioned in #552. The `setup.py` moves the tests there during install anyway.

### CODE REVIEW

**Comment by wd15** [comment]
Just a note that you can ensure that the canonical way of running the tests is just "py.test" without addition arguments by adding the configuration to the setup.cfg as well as including and excluding tests. For example, pymks uses [this](https://github.com/materialsinnovation/pymks/blob/master/setup.cfg#L23).  Sfepy doesn't have a setup.cfg so might be handy to have one. Also, I'm not sure that setup.cfg is the latest and greatest way for python packaging, but it's what I've been using. It seems that the pyproject.toml is being used a lot.

**Comment by rc** [comment]
Thanks @wd15!

**Comment by rc** [comment]
The current source directory structure (only relevant shown) is
```
examples/
meshes/
sfepy/
tests/
```
while after the installation it is
```
sfepy/
...examples/
...meshes/
...tests/
```
With this setup, I am getting test failures (such as `ModuleNotFoundError: No module named 'examples.linear_elasticity'`) in examples that import other examples when testing the installed package.
I can "fix" that by adding `~/.local/lib/python3.8/site-packages/sfepy/` to `PYTHONPATH`, but that is not a solution.

So to fix it properly, I am considering moving both `tests/` and `examples/` under `sfepy/`, while leaving `meshes/` alongside `sfepy/` in the sources (those are OK thanks to `sfepy.data_dir` variable). I plan to use absolute imports, e.g. `from sfepy.examples.linear_elasticity.its2D_1 import ...` to avoid possible ambiguities in example names.

What do you think?

**Comment by rc** [comment]
I guess it is almost ready. Any idea what causes the AppVeyor failure? Or would it be better to use github actions?

**Comment by vlukes** [comment]
Would it be possible to use a more user-friendly way of calling the tests? E.g. `./simply.py --tests`? Or `./sfepy-run tests`?

**Comment by rc** [comment]
Actually you can do (in the top source directory)
```
export PYTHONPATH=.
```
and then just
```
pytest sfepy/tests/
```
The way in the docs is a general one that works also for an installed sfepy version.

I would like to add a sfepy-test entry point in another PR, that would make sfepy pip-installable. Not sure when I get to it, though.

File: .appveyor.yml [hunk 1]
```diff
@@ -38,9 +38,9 @@ environment:
   matrix:
 
     - PYTHON: "C:\\Python37_64"
-      PYTHON_VERSION: "3.7"
+      PYTHON_VERSION: "3.9"
```
Comment (vlukes) [inline .appveyor.yml:41]: What is the reason to require Python 3.9?

**Review by vlukes — APPROVED** [review]

File: .appveyor.yml (hunk 1 above)
Comment (rc) [reply .appveyor.yml:41]: With 3.7 the tests did not work, so I tried updating. No help, it seems there are some problems related to using conda.

Comment (vlukes) [reply .appveyor.yml:41]: This is a very strict requirement. The latest LTS Ubuntu 20.4 has Python 3.8.

Comment (rc) [reply .appveyor.yml:41]: We can change it to 3.8, as it does not work anyway :)

**Review by vlukes — APPROVED** [review]

**Comment by rc** [comment]
@wd15 I have marked several of the tests suggested as slow in #552 with the `slow` marker. I will merge this PR now as it is to move on, we can discuss further improvements in future PRs. Maybe you could use pytest's command line arguments to restrict the tests run to your exact needs?

**Comment by wd15** [comment]
Great work @rc! What is the correct way to run the tests now? Would running

    $ python -c "import sfepy; sfepy.test()"

for an installed version run a fast version of the tests? What about for a cloned version?

**Comment by rc** [comment]
Have a look [here](https://sfepy.org/doc-devel/installation.html#running-automated-test-suite). Let me know if the docs are not clear enough. I shamelessly copied the pymks `test()` function :)

**Comment by vlukes** [comment]
I suggest adding a note on how to run a single test.

**Comment by rc** [comment]
Thanks for the feedback, I tried to improve the docs in #814.

**Pull request merged**